src/doctypes/conflicts.spec.js

```diff
@@ -0,0 +1,99 @@
+import schema from './schema'
+import CozyClient from 'cozy-client'
+import { saveDocUpdateHandlingConflicts } from './conflicts'
+
+const makeConflictError = () => ({
+  message: JSON.stringify({
+    error: 'conflict'
+  })
+})
+
+describe('conflict resolution', () => {
+  const setup = () => {
+    const client = new CozyClient({
+      schema: schema
+    })
+    let i = 0
+    jest.spyOn(client, 'save').mockImplementation(async doc => {
+      if (i === 0) {
+        i++
+        throw makeConflictError()
+      } else {
+        return doc
+      }
+    })
+
+    const originalSettings = {
+      _id: '1234',
+      _type: 'io.cozy.bank.settings',
+      categoryBudgetAlerts: [
+        {
+          id: 0,
+          maxThreshold: 10,
+          categoryId: '400100'
+        }
+      ]
+    }
+
+    const serverSettings = {
+      ...originalSettings,
+      categoryBudgetAlerts: [
+        {
+          ...originalSettings.categoryBudgetAlerts[0],
+          lastNotificationAmount: 120, // server has changed lastNotificationAmount
+          lastNotificationDate: '2019-08-07T12:00' // server has changed lastNotificationDate
+        }
+      ]
+    }
+
+    jest.spyOn(client, 'query').mockResolvedValue({
+      data: serverSettings
+    })
+
+    return { client, originalSettings }
+  }
+
+  it('should correctly handle a conflict (concurrent changes)', async () => {
```

Comment:
Nit: Instead of putting context of the test case in parenthesis, I like to do something like:

```js
describe('conflict resolution', () => {
  describe('when concurrent changes', () => {
     it('should correctly handle a conflict', () => { /* ... */ })
  })

  describe('when deleting an element in an array', () => {
    it('should correctly handle a conflict', () => { /* ... */ })
  })
})
```

It's easier to add a test case for a given context and gives better output in jest summary IMHO.